src/main/java/org/apache/sysds/runtime/transform/encode/MultiColumnEncoder.java

```diff
@@ -72,11 +96,60 @@ public MatrixBlock encode(FrameBlock in) {
 	}
 
 	public void build(FrameBlock in) {
-		for(ColumnEncoder columnEncoder : _columnEncoders)
-			columnEncoder.build(in);
+		build(in, 1);
+	}
+
+	public void build(FrameBlock in, int k) {
+		if(MULTI_THREADED && k > 1) {
+			buildMT(in, k);
+		}
+		else {
+			for(ColumnEncoder columnEncoder : _columnEncoders)
+				columnEncoder.build(in);
+		}
 		legacyBuild(in);
 	}
 
+	private void buildMT(FrameBlock in, int k) {
+		int blockSize = BUILD_BLOCKSIZE <= 0 ? in.getNumRows() : BUILD_BLOCKSIZE;
+		List<Callable<Integer>> tasks = new ArrayList<>();
```

Comment:
Thats a good point.
This should not be a problem, since the encoders are independent we can just call the encoders with the blocksize we need.
So in the future we might get a array of blocksizes which we then just need to match